### PR TITLE
Test speedup 2022 selenium firefox

### PR DESCRIPTION
**Pull request by jirik**
Part of issue #

- [ ] Tests
- [ ] Layman Test Client
- [ ] Changlelog
- [ ] Documentation

Optionally
- Add dependency to doc/dependencies.md

### CODE REVIEW

File: docker/Pipfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ requests = "*"
 owslib = "==0.22.0" # https://github.com/geopython/OWSLib/issues/762
 jsonschema = "*"
 flower = "*"
-selenium = "*"
+selenium = ">=4"
```
Comment (index-git) [inline docker/Pipfile:26]: We should document this in dependencies and changelog.

File: src/layman/layer/client_test.py [hunk 2]
```diff
@@ -23,25 +23,31 @@ def clear_country_chunks():
 
 
 @pytest.fixture(scope="module")
-def chrome():
-    chrome_options = Options()
-    chrome_options.add_argument("--headless")
-    chrome_options.add_argument("--no-sandbox")
-    desired_capabilities = DesiredCapabilities.CHROME
+def firefox():
+    print(f"firefox fixture START")
```
Comment (index-git) [inline src/layman/layer/client_test.py]: Remove prints from this method.

File: src/layman/layer/client_test.py [hunk 3]
```diff
@@ -23,25 +23,31 @@ def clear_country_chunks():
...
+    desired_capabilities = DesiredCapabilities.FIREFOX
     desired_capabilities['loggingPrefs'] = {'browser': 'ALL'}
-    chrome = webdriver.Chrome(
-        options=chrome_options,
+    print(f"firefox fixture 1")
+    firefox = webdriver.Firefox(
+        options=firefox_options,
         desired_capabilities=desired_capabilities,
+        service_log_path="/code/tmp/artifacts/gecko_service_log_path.txt",
+        log_path="/code/tmp/artifacts/gecko_log_path.txt",
```
Comment (index-git) [inline src/layman/layer/client_test.py:35]: In my experience, "log_path" is not used.

File: src/layman/layer/client_test.py [hunk 4]
```diff
@@ -23,25 +23,31 @@ def clear_country_chunks():
...
+    yield firefox
+    firefox.close()
+    firefox.quit()
 
 
 @pytest.mark.test_client
 @pytest.mark.usefixtures('ensure_layman', 'clear_country_chunks')
-def test_post_layers_chunk(chrome):
+def test_post_layers_chunk(firefox):
+    print(f"test_post_layers_chunk START")
```
Comment (index-git) [inline src/layman/layer/client_test.py]: Remove prints from this method.

File: src/layman/layer/client_test.py [hunk 5]
```diff
@@ -180,17 +177,15 @@ def test_patch_layer_chunk(firefox):
         raise exc
     firefox.save_screenshot('/code/tmp/artifacts/client-patch-layers-4.png')
 
-    entries = firefox.get_log('browser')
-    performance_entries = json.loads(firefox.execute_script("return JSON.stringify(window.performance.getEntries())"))
```
Comment (index-git) [inline src/layman/layer/client_test.py]: "performance_entries" are added in this PR and later deleted in ti.

File: src/layman/map/filesystem/thumbnail.py [hunk 6]
```diff
@@ -85,30 +85,48 @@ def generate_map_thumbnail(workspace, mapname, editor):
...
-    chrome = webdriver.Chrome(
-        options=chrome_options,
+    firefox_options = Options()
+    firefox_options.headless = True
+    desired_capabilities = DesiredCapabilities.FIREFOX
+    desired_capabilities['loggingPrefs'] = {'browser': 'ALL'}
+    firefox = webdriver.Firefox(
+        options=firefox_options,
         desired_capabilities=desired_capabilities,
+        service_log_path="/code/tmp/artifacts/gecko_service_log_path.txt",
```
Comment (index-git) [inline src/layman/map/filesystem/thumbnail.py:95]: We should change service_log_path and remove log_path.

File: src/layman/map/filesystem/thumbnail.py [hunk 7]
```diff
@@ -85,30 +85,48 @@ def generate_map_thumbnail(workspace, mapname, editor):
...
+        log_path="/code/tmp/artifacts/gecko_log_path.txt",
     )
-    chrome.set_window_size(500, 500)
 
-    chrome.get(timgen_url)
-    entries = chrome.get_log('browser')
+    current_app.logger.info(f"After creating webdriver.Chrome")
```
Comment (index-git) [inline src/layman/map/filesystem/thumbnail.py:99]: Remove logs.

**Review by index-git — CHANGES_REQUESTED** [review]

File: .github/workflows/separated_tests.yml [hunk 8]
```diff
@@ -80,5 +80,6 @@ jobs:
         with:
           name: test_artifacts
           path: |
+            sudo chown -R $(id -un) layman_data_test
```
[the remark this answers is not in the document]
Comment (jirik) [reply .github/workflows/separated_tests.yml:83]: Not so easy to test, it's in case of failure()

**Pull request closed without merging**